includes/wikia/nirvana/WikiaResponse.class.php

```diff
@@ -304,7 +303,14 @@ public function setHeader( $name, $value, $replace = true ) {
 	 * @param string $policy caching policy (either private or public)
 	 */
 	public function setCachePolicy($policy) {
-		$this->cachingPolicy = $policy === self::CACHE_PRIVATE ? self::CACHE_PRIVATE : self::CACHE_PUBLIC;
+		if ( in_array( $policy, [
+				self::CACHE_PRIVATE,
+				self::CACHE_PUBLIC,
+				self::CACHE_ANON_PUBLIC_USER_PRIVATE,
+			] )
+		) {
+			$this->cachingPolicy = $policy;
+		}
```

Comment:
what about error handling in case `$policy` is not valid? I would log an error/exception at least.